services/userService.js

```diff
@@ -0,0 +1,43 @@
+const db = require('./dbConnect');
+const UserService = {};
+
+UserService.createUser = (uid, username, email) => {
+    const sql = `
+        INSERT INTO users (uid, username, email)
+        VALUES ($[uid], $[username], $[email])
+        RETURNING id;
+    `;
+    return db.one(sql, { uid, username, email });
+};
+
+UserService.readUserByUid = uid => {
+    const sql = `
+        SELECT *
+        FROM users
+        WHERE users.uid = $[uid]
+    `;
+
+    return db.one(sql, { uid });
+};
+
+UserService.readUserById = id => {
+    const sql = `
+        SELECT *
+        FROM users
+        WHERE users.id = $[id]
+    `;
+
+    return db.one(sql, { id });
+};
+
+UserService.readUserByEmail = email => {
+    const sql = `
+        SELECT *
+        FROM users
+        WHERE users.email = $[email]
+    `;
+
+    return db.one(sql, { email });
+};
+
+module.exports = UserService;
```

Comment:
I really like that you're splitting up this up into small, separate functions. Really great adherence to the Single Responsibility Principle _(if you haven't heard of it before, +1 for following it anyway!)_.

One suggestion: there's no real need to create a `UserService` object and assign methods on it. You could write this file like this:

```js
export const readUserByEmail = email => {
    // ...
};
```

I found it a little confusing at first reading through this file and wondering _("well, who takes in the `UserService` object itself?")_ until I saw it was only used as `module.exports`.

Another point in favor: setting `module.exports` is starting to be discouraged now, in favor of [ES Modules](https://nodejs.org/api/esm.html#esm_ecmascript_modules)

Reply:
Hmm, that docs page is kind of overly Node-y technical focused, this one is nicer: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export

Reply:
This is so cool, we're used to writing module.exports to export all the methods, so that in the router file we can easily call them to make whatever queries we need to create. Would a good approach be to export each individual function?

Reply:
Yup! The two do the same things.

This:

```js
export const wat = () => 1;
```

Is functionally about the same as:

```js
module.exports.wat = () => `;
```